### DISCUSSION TITLE
React Native and React DOM should not be in `peerDependencies`

### DISCUSSION THREAD

**Discussion by ingvaldlorentzen** [TanStack/query#3993] [General]
I made a [comment on the PR related to this](https://github.com/TanStack/query/pull/3786), but thought it might be better as it's own discussion. 

Since `npm` installs optional dependencies by default, I think `react-query` should remove `react-native` and `react-dom` from it's `peerDependencies`. 

Obviously this is not optimal, but until this is fixed in `npm` I think it's better than the current situation. There is a [discussion on this](https://github.com/npm/feedback/discussions/225) in the `npm/feedback` repo.

Take a "regular" webproject that wants to use `react-query`, by default `react-native` will be installed, which is not only unnecessary, it also quickly creates build issues as `react-native` has very different requirements from a webproject. 

It also limits other dependencies in the project based on the requirements `react-native` has. E.g. using React `v18.2.0` is not possible in a webproject with `react-query` becasue `react-native` does not support `v18.2.0`

Would love to hear other suggestions on how to solve this, both in this project and in projects consuming it.

**Comment by TkDodo** [comment]
I agree that this is suboptimal, but from what I can see, we're correctly defining it as an optional peer dependency, and npm shouldn't install it ?
